### PR TITLE
Don't validate named ranges when loading file

### LINKED ISSUE #1011: When workbook contains NamedRange starting with "[" the workbook can't be opened. Exception message is "The named range name '\y' does not begin with a letter or an underscore"
**Do you want to request a *feature* or report a *bug*?**
- [x ] Bug

**Version of ClosedXML**
0.93.1

**What is the current behavior?**

When opening the workbook, ClosedXml crashes.

**What is the expected behavior or new feature?**

It's possible to create a named range starting with "[" in excel. ClosedXml should be able to open a spreadsheet using this format.

**Did this work in previous versions of our tool?  Which versions?**

No

## Reproducibility

**Code to reproduce problem:**
```c#
public void Main()
{
   var wb = new XLWorkbook("Test Workbook.xlsx");
}
```
- [x] I attached a sample spreadsheet.  (You can drag files on to this issue)
[Test Workbook.xlsx](https://github.com/ClosedXML/ClosedXML/files/2425615/Test.Workbook.xlsx)

### PR DESCRIPTION
**Pull request by igitur**
Fixes #1011 
Supercedes #1012 

We already don't validate the named range address when loading from file. Now also don't validate the name. 

But we still enforce the rules when adding named ranges from the ClosedXML API

### CODE REVIEW

File: ClosedXML/Excel/NamedRanges/XLNamedRange.cs [hunk 1]
```diff
@@ -76,6 +87,11 @@ public String Name
             }
         }
 
+        internal void SetNameWithoutValidation(string value)
```
Comment (Pankraty) [inline ClosedXML/Excel/NamedRanges/XLNamedRange.cs]: Why not `private`?

File: ClosedXML/Excel/NamedRanges/XLNamedRange.cs [hunk 2]
```diff
@@ -13,22 +13,33 @@ internal class XLNamedRange : IXLNamedRange
         internal XLWorkbook Workbook => _namedRanges.Workbook;
 
         public XLNamedRange(XLNamedRanges namedRanges, String rangeName, String range, String comment = null)
-            : this(namedRanges, rangeName, comment)
+            : this(namedRanges, rangeName, validateName: true, comment)
```
Comment (Pankraty) [inline ClosedXML/Excel/NamedRanges/XLNamedRange.cs]: `: this(namedRange, rangeName, validateName: true, range, comment)`
- this way you can leave the constructor's body empty.

**Comment by Pankraty** [comment]
The approve button is not visible once I submitted the review. Everything looks fine for me.

**Comment by igitur** [comment]
I'm still trying to clarify a few things in #990 . Notice that the test that I added overrides the default fr-FR decimal symbol. But maybe that shouldn't happen.

**Pull request merged**